internal/wasm/sdk/opa/pool.go

```diff
@@ -222,6 +222,40 @@ func (p *pool) SetDataPath(path []string, value interface{}) error {
 }
 
 func (p *pool) RemoveDataPath(path []string) error {
```

Comment:
Are we sure this function as well as `SetDataPath` are safe for concurrent access? It looks like we're relying on the fine-grained locking from the other functions however the rest of the exported functions on the pool all use coarse-grained locking. I'm a bit concerned about concurrent access here.

Reply:
As sure as I am about the rest of the sdk code 😅   Did you have some specific concern or issue you saw?

It's the same concurrency model as when the VM's get new data/policy.  The public `SetPolicyData` one uses the course locking to cover the initialization phase of the pool, and for getting a copy of the current set of data+policy (which the `activate` helper guards with the same lock while setting). The flow where we acquire each VM and update, or throw them away, remains the same. So if anything its consistently wrong 🤣 

I'm pretty confident that once we have acquire the VM from `wait` we are safe. The main points to be worried about are the interactions between `wait`, `remove`, and the normal `Acquire`/`Release` workflows. I can take some time to go through and audit those... we don't really have any test coverage for them.